jans-cedarling/cedarling/src/jwt/decoding_strategy.rs

```diff
@@ -5,140 +5,197 @@
             DecodingStrategy::WithValidation {
                 key_service,
                 supported_algs,
-            } => decode_and_validate_jwt(jwt, iss, aud, req_sub, supported_algs, key_service),
+            } => decode_and_validate_jwt(
+                jwt,
+                iss,
+                aud,
+                sub,
+                validate_nbf,
+                validate_exp,
+                supported_algs,
+                key_service,
+            ),
         }
     }
-}
 
-/// Trait implementation for extracting claims without validation.
-impl ExtractClaims for DecodingStrategy {
-    /// Extracts the claims from a JWT token without validation.
-    fn extract_claims<T: DeserializeOwned>(&self, jwt_str: &str) -> Result<T, Error> {
+    /// Extracts the claims from a JWT token without performing validation.
+    ///
+    /// This method uses a default insecure validator that skips signature
+    /// validation and other checks (e.g., expiration). Only use in trusted environments.
+    ///
+    /// # Errors
+    /// Returns an error if the claims cannot be extracted.
+    pub fn extract_claims<T: DeserializeOwned>(&self, jwt_str: &str) -> Result<T, Error> {
         let mut validator = jwt::Validation::default();
         validator.insecure_disable_signature_validation();
+        validator.required_spec_claims.clear();
         validator.validate_exp = false;
         validator.validate_aud = false;
         validator.validate_nbf = false;
 
         let key = jwt::DecodingKey::from_secret("some_secret".as_ref());
 
-        let claims = jwt::decode::<T>(jwt_str, &key, &validator)
-            .map_err(Error::Validation)?
+        let claims = jwt::decode::<T>(&jwt_str, &key, &validator)
+            .map_err(|e| Error::ValidationError(e.to_string()))?
             .claims;
 
         Ok(claims)
     }
 }
 
-/// Decodes and validates a JWT token based on supported algorithms and a key service.
+/// Decodes and validates a JWT token using supported algorithms and a key service.
+///
+/// # Arguments
+/// - `jwt` - The JWT string to decode.
+/// - `iss` - Optional expected issuer for validation.
+/// - `aud` - Optional expected audience for validation.
+/// - `sub` - Optional expected subject for validation.
+/// - `validate_nbf`: A boolean indicating whether to validate the "not before" claim.
+/// - `validate_exp`: A boolean indicating whether to validate the expiration claim.
+/// - `supported_algs` - A reference to a vector of supported algorithms for validation.
+/// - `key_service` - A reference to a `KeyService` to retrieve keys for signature validation.
 ///
 /// # Errors
-/// - Returns an error if the JWT header specifies an unsupported algorithm.
-/// - Returns an error if validation fails due to signature mismatch or claim validation failure.
+/// Returns an error if the token uses an unsupported algorithm or if validation fails.
 fn decode_and_validate_jwt<T: DeserializeOwned>(
```

Comment:
Can we create structure for this 8 arguments?

Reply:
it's used internally... it's only going to make more code without much benefit...

Reply:
fixed in [77823a6](https://github.com/JanssenProject/jans/pull/9939/commits/77823a65bf85c97837618198986fd53f7c65b68e)